src/tool/hpcrun/gpu/amd/roctracer-api.c

```diff
@@ -577,6 +577,17 @@ roctracer_init
   HPCRUN_ROCTRACER_CALL(roctracer_enable_domain_callback, (ACTIVITY_DOMAIN_ROCTX, roctracer_subscriber_callback, NULL));
 }
 
+void
+roctracer_flush
+(
+ void* args,
+ int how
+)
+{
+  HPCRUN_ROCTRACER_CALL(roctracer_flush_activity_expl, (NULL));
+}
```

Comment:
@jmellorcrummey I have the feeling that we should also move the call to `gpu_application_thread_process_activities()` in `roctracer_fini` to `roctracer_flush`. Otherwise, child threads will flush roctracer activities without process them?

Reply:
agreed. done.